test/gcs-resumable-upload.ts

```diff
@@ -67,6 +67,8 @@ const RESUMABLE_INCOMPLETE_STATUS_CODE = 308;
 /** 256 KiB */
 const CHUNK_SIZE_MULTIPLE = 2 ** 18;
 const queryPath = '/?userProject=user-project-id';
+const X_GOOG_API_HEADER_REGEX =
+  /^gl-node\/[0-9]+\.[0-9]+\.[-.\w]+ gccl\/[0-9]+\.[0-9]+\.[-.\w]+ gccl-invocation-id\/[\w]{8}-[\w]{4}-[\w]{4}-[\w]{4}-[\w]{12}$/;
```

Comment:
The `gl-node/...` and `gccl/...` portions may not match for tagged semver preview/beta releases.

Here's a suggestion:

```suggestion
  /^gl-node\/(?<nodeVersion>[^W]+) gccl\/(?<gccl>[^W]+) gccl-invocation-id\/(?<gcclInvocationId>[^W]+)$/
```

Which will allow easy access to named groups for testing:
```ts
const match = X_GOOG_API_HEADER_REGEX.exec(...);

assert(match);

// match.groups.nodeVersion
// match.groups.gccl
// match.groups.gcclInvocationId
```

Reply:
^this suggestion is optional